### ISSUE TITLE
Gradle plugin checks for connected devices at configuration time

### ISSUE THREAD

**Issue by jschear** [airbnb/okreplay#91]
Hi! I've noticed that when gradle is configuring my `app` project (which has the OkReplay gradle plugin applied), I see this output in the console if I don't have any android devices connected:

```
> Configure project :app  
No connected devices!
No connected devices!
No connected devices!
```
(We have three build variants.)

I think what's happening is that when the OkReplay tasks are setup, a `DeviceBridge` [is created](https://github.com/airbnb/okreplay/blob/master/okreplay-gradle-plugin/src/main/kotlin/okreplay/OkReplayPlugin.kt#L69), which [eagerly calls `ConnectedDeviceProvider#init()`](https://github.com/airbnb/okreplay/blob/master/okreplay-gradle-plugin/src/main/kotlin/okreplay/DeviceBridge.kt#L14), which in turn uses `adb` to check for connected devices.

I think this call to `init()`should probably be moved out of the configuration stage, and into the tasks themselves (it looks like this is what the Android Gradle Plugin does with [`DeviceProviderInstrumentTestTask`](https://android.googlesource.com/platform/tools/base/+/mirror-goog-studio-master-dev/build-system/gradle-core/src/main/java/com/android/build/gradle/internal/tasks/DeviceProviderInstrumentTestTask.java#192) by using the newer `DeviceProvider.use` API). I imagine this will cut down configuration time a tiny bit for all of our builds.

If y'all agree, I'm happy to open a PR to do so.

**Comment by technoir42** [comment]
Should be fixed by #99

**Comment by rossbacher** [comment]
Fixed bu #99
